gobblin-temporal/src/main/java/org/apache/gobblin/temporal/joblauncher/GobblinTemporalJobLauncher.java

```diff
@@ -65,10 +72,13 @@
 @Alpha
 public abstract class GobblinTemporalJobLauncher extends GobblinJobLauncher {
   private static final Logger log = Workflow.getLogger(GobblinTemporalJobLauncher.class);
+  private static final int TERMINATION_TIMEOUT_SECONDS = 3;
```

Comment:
I would timeout at least 30 seconds, 3 is quite aggresive and the consequence of an abrupt termination can be quite high currently until we have pipelines cleaning after prior executions in Temporal

Reply:
The application master gets killed within 5 seconds. We only have upto 5 seconds to perform the cancellation. 3 seems to be a safe number for this wait, given that there might be other things which take up some time.